editor/src/clj/editor/slice9.clj

```diff
@@ -220,10 +221,20 @@
         x-steps [0.0 ^double (get slice9 0) (- box-width ^double (get slice9 2)) box-width]
         y-steps [0.0 ^double (get slice9 3) (- box-height ^double (get slice9 1)) box-height]
         xy-box-coords (ranges->box-corner-coords (steps->ranges x-steps) (steps->ranges y-steps))
-        non-empty-xy-box-coords+uv-boxes (into []
-                                               (filter (fn [[[x0 y0 x1 y1] _uv-box]]
-                                                         (and (not= x0 x1) (not= y0 y1))))
-                                               (map vector xy-box-coords uv-boxes))
+        empty-xy-box-coords+uv-boxes (mapv pair xy-box-coords uv-boxes)
+
+        non-empty-xy-box-coords+uv-boxes
+        (or (coll/not-empty
+              (filterv
+                (fn [[[x0 y0 x1 y1] _uv-box]]
+                  (and (not= x0 x1) (not= y0 y1)))
+                empty-xy-box-coords+uv-boxes))
+            ;; We need *some* geometry, or it will eventually result in invalid
+            ;; buffers that will likely crash the GL driver. When all 9 boxes
+            ;; are empty, use the center slice. This can happen with a manual
+            ;; size of zero.
+            [(empty-xy-box-coords+uv-boxes 4)])
+
```

Comment:
Second issue: If the manual `size` was zero, we'd generate no geometry. Our callees expect some geometry for populating the vertex and buffers.